src/core/providers/ogr/qgsgeopackageproviderconnection.cpp

```diff
@@ -380,3 +408,14 @@ QList<QVariantList> QgsGeoPackageProviderConnection::executeGdalSqlPrivate( cons
   return results;
 }
 
+
+QList<QgsVectorDataProvider::NativeType> QgsGeoPackageProviderConnection::nativeTypes() const
+{
+  QList<QgsVectorDataProvider::NativeType> types;
+  QgsVectorLayer vl { uri(), QStringLiteral( "temp_layer" ), QStringLiteral( "ogr" ) };
+  if ( ! vl.isValid() || ! vl.dataProvider() )
+  {
+    throw QgsProviderConnectionException( QObject::tr( "Error retrieving native types for %1: %2" ).arg( uri() ).arg( vl.dataProvider()->errors().join( '\n' ) ) );
+  }
```

Comment:
@elpaso in the case of `! vl.dataProvider()` you're still trying to access `vl.dataProvider()->errors()` which causes QGIS to crash.

Reply:
Yeah, of course :/ I'll fix it. Thank you for notifying.